### ISSUE TITLE
Unusually increased Hill height at a few points in the metadynamics simulation

### ISSUE THREAD

**Issue by Praveen-2k21** [Colvars/colvars#514]
Hello everyone, 

I have been running metadynamics simulations of an organic solute in a membrane. I started my simulations with an initial Gaussian hill height of 0.3 kcal/mol (given by the keyword hillWeight in the configuration file) and according to theory, as the simulation converges, the hill height would reduce to zero. After about 0.8 microseconds of simulation, I noticed that at a few instances, the hill height is "nan" and a few instances where the hill height was greater than 0.3 kcal/mol (the initial hill height). It would be great if you could share any insights on why this could be happening and what I could do to correct it?

**Comment by giacomofiorin** [comment]
Hi @Praveen-2k21 can you report which variables you used including the config file?

Please keep in mind that in metadynamics the hill height is always finite, and it won't go to zero unless you are using well-tempered metadynamics (Barducci et al, 2008). If you are basing your reasoning upon a different theory, can you specify it? 

Also, consider how quickly you're adding the bias, i.e. about 0.6 kcal/mol per ps if you are using 2 fs as the integration step. That's really quick, considering that the relevant membrane relaxation times are nanoseconds at the very least.

I would like to see more details as to where the "nan" was reported to exclude possible bugs in the code. But either way, the way you described the simulation setup I don't think you can get a stable simulation, let alone a statistically converged one.

Lastly, whether you get numbers that make sense or not, you should move forward only after having visualized the atomic trajectory. What is happening to your solute and the membrane during this run?

**Comment by Praveen-2k21** [comment]
Hi Giacomo, 

Many thanks for your quick response. 

yes, I am using Well-tempered Metadynamics. Apologies I didn't add WT-Metadynamics when I posted the issue. 

I am using a 2 fs/step timestep. I am having hill weight of 0.3 kcal/mol and frequency of 500 steps (which 1 ps) so the rate would be 0.3 kcal/mol. 

Thank you for sharing the information of relaxation times of membranes. 

I am listing below the key variables, please let me know if I have to share any other information

Temperature: 303.15 K
biasTemperature = 916.5 K #bias factor of 4
newHillFrequency = 500
hillWeight = 0.3
hillwidth=2.0
timestep = 2.0 fs/step

Thrermostat: Stochastic velocity rescale thermostat
stochRescalePeriod =1

I am able to see the solute diffuse through the membrane but the entire system seems to have very high velocity. The "nan" I have noticed occurs when the simulation crashes and most of the time the error I get in the log file is certain atoms are moving with very high velocities. 

Another part which is confusing me there a few times in the welltempered Metadynamics simulation is where the hill height is higher than the initial hill height (the simulation is not crashing though)  

Thank you so much for your help! 

Please find attached the hills file. 
[Hills3.txt](https://github.com/Colvars/colvars/files/10680701/Hills3.txt)

**Comment by giacomofiorin** [comment]
@Praveen-2k21 Thanks for specifying the MD settings, I corrected my wrong estimate of your bias rate. That remains very high, though.

In the file attached I see that the hill height is either 0.3 kcal/mol or much lower numbers where they are significantly depressed by the well-tempered algorithm. I can't see intermediate numbers, but possibly this because your grid is rather coarse, and the hills are discretized on it. So basically, I am unable to see any numerical issues with the computation.

I would take a step back and simplify the configuration, first by testing that the CV's definition (not included here!) is reliable without biasing forces, then add a simple metadynamics bias (non WT) that doesn't blow up your system, and if you want a smoother PMF consider WT or manual averaging.

I'm closing this because it looks like a protocol issue. If you encounter bugs, of course, feel free to open another issue.
